### ISSUE TITLE
[NEWS] Dodaliśmy nowy filtr wyszukiwania dla operacji billingowych / We have introduced a new search filter for billing operations

### ISSUE THREAD

**Issue by MartaNowaczyk** [allegro/allegro-api#5881]
Udostępniliśmy nowy filtr wyszukiwania **order.id** dla [GET /billing/billing-entries](https://developer.allegro.pl/documentation/#operation/getBillingEntries), dzięki temu pobierzesz operacje billingowe powiązane z danym zamówieniem.

-----
We have introduced a new **order.id** search filter for[ GET /billing/billing-entries](https://developer.allegro.pl/documentation/#operation/getBillingEntries), thanks to this you will return all billing operations associated with the passed order.

**Comment by allegrofan** [comment]
filtr nie zwraca opłat za przesyłkę. Np smart do paczkomatu 0,99zł

**Comment by PrzemyslawLukanowski** [comment]
Nie udało mi się powtórzyć tego błędu. Filtr uwzględnia ten rodzaj operacji (HB4). Możesz podać trace-id, gdzie nie otrzymałeś tego rodzaju operacji?

**Comment by PrzemyslawLukanowski** [comment]
Tak, działa, przed chwilą to sprawdziłem. Być może nie masz na koncie żadnych operacji związanych z tym kodem? Możesz to sprawdzić w zakładce[ Rozliczenia z Allegro](https://allegro.pl/moje-allegro/moje-konto/rachunki/rozliczenia-z-allegro?sellerId=62799754).

**Comment by kepesian** [comment]
HB4 to Allegro paczkomaty Inpost czyli rozumiem że w rozliczeniach widoczne również  jako (Allegro Paczkomaty InPost - w ramach Uslugi Allegro Smart); HB2- Przesyłka Poczta Polska  -  rozumiem, że pod to podchodzi z rozliczeń opis ( Allegro Pocztex Kurier 48 - w ramach Uslugi Allegro Smart) , czy może się mylę i inny filtr tutaj nalezy zastosować ? ( Zwraca mi puste tablice) mimo, że w rozliczeniu na koncie widoczne są. Dodatkowo, gdy nie ustawię pola type.id , to w response otrzymiuję tablice, w których nie ma żadnego id typu który byłby zgodny z jakąkolwiek wysyłką, a przeglądam np. ofertę z okresu 4 miesięcy.

**Comment by PrzemyslawLukanowski** [comment]
Tak, to prawidłowy typ. Całą listę wszystkich dostępnych typów możesz pobrać za pomocą [GET /billing/billing-types](https://developer.allegro.pl/documentation#operation/getBillingTypes). Zarówno na stronie, jak i w API, przy wpisie znajduje się numer zamówienia powiązany z opłatą. Czy możesz go użyć jako wartości w parametrze order.id? Czy w takim przypadku również nie otrzymujesz opłaty, którą widzisz w zakładce [Rozliczenia z Allegro](https://allegro.pl/moje-allegro/moje-konto/rachunki/rozliczenia-z-allegro?sellerId=62799754)? Jeśli tak, to czy możesz przesłać zrzut ekranu, który obejmuje wyłącznie ten odcinek dotyczący opłaty, której nie zwracamy w API oraz trace id wywołania?

**Comment by kepesian** [comment]
w rozliczeniach dla billingTypes=HB2 lub billingTypes=HB4 -  dane sa pokazywane prawidłowo -  są opłaty z tych tyułów, ale z API dostaje puste tablice dla tych filtrów. Dla innycch filtróe "SUC", "FEA""FSF" odpowiedzi są poprawne. Nie wiem jak jest dla ITR i DDP, bo nie mam takich

**Comment by PrzemyslawLukanowski** [comment]
Prosiłbym w takim razie dane, o których wspomniałem oraz przykładowe trace id wywołania, gdzie nie otrzymujesz wpisów H2 i H4.

**Comment by kepesian** [comment]
OK, filtry działają, ale wyłącznie dla podania order.id, gdy podaje offer.id  - nie uzyskuję danych o opłatach pobranych za wysyłki choć filtrem jest typ operacji.

**Comment by kepesian** [comment]
Reasumują, gdy parametry są 
 $param = array(
                    "offset" =>  $offset, // jakiś offset
                    "occurredAt.gte" => $DateReport_start 
                    "occurredAt.lte" => $DateReport_end
                    "offer.id" => $OfferNumber, // moja oferta 
                    "type.id" => "HB4",
                );
  $data =  $api->getDirect('/billing/billing-entries',  $param);

zwracany jest pusty rekord - nie ma opłat  

Jeżeli jednak dane są zamiast offer jest konkretny order 
 $param = array(
                    "offset" =>  $offset, // jakiś offset
                    "occurredAt.gte" => $DateReport_start 
                    "occurredAt.lte" => $DateReport_end
                    "order.id" => $OrderId // jakiś konkretny order
                    "type.id" => "HB4",
                );
  $data =  $api->getDirect('/billing/billing-entries',  $param);

otrzymuję talice z jednym rekordem ( to oczywiste bo dla jednego zamówienia jest jedna opłata za przesyłkę). 

Pytanie brzmi jak pobrać opłaty związane ze sprzedażą produktów z konkretnej oferty, czyli jak opłaty pobrane filterm np."SUC"

 $param = array(
                    "offset" =>  $offset, // jakiś offset
                    "occurredAt.gte" => $DateReport_start 
                    "occurredAt.lte" => $DateReport_end
                    "offer.id" => $OfferNumber, // moja oferta 
                    "type.id" => "SUC",
                );
  $data =  $api->getDirect('/billing/billing-entries',  $param);

który zwraca to co powinno, tablice z tyloma rekordemi ile razy wystapiła opłata SUC.

**Comment by kepesian** [comment]
OK, rozumiem, czyli jeżeli mam 1000 wysyłek, to oaby pobrać sumę opłat HB4, pownieniem 1000 razy wysłać requesta o opłatę HB4 z order.id jako parametrem ?

**Comment by PrzemyslawLukanowski** [comment]
> Pytanie brzmi jak pobrać opłaty związane ze sprzedażą produktów z konkretnej oferty, czyli jak opłaty pobrane filterm np."SUC"

Dokładnie tak, jak to przedstawiłeś, czyli za pomocą filtrów offer.id i type.id + opcjonalnie data wystąpienia opłaty i offset.

> OK, rozumiem, czyli jeżeli mam 1000 wysyłek, to oaby pobrać sumę opłat HB4, pownieniem 1000 razy wysłać requesta o opłatę HB4 z order.id jako parametrem ?

Możesz ominąć parametr order.id i w ten sposób zwrócimy wszystkie opłaty typu HB4 w ramach prowadzonej sprzedaży. W ten sposób również wyliczysz sumę opłat.

**Comment by kepesian** [comment]
OK, dziękiję - zrobie to zwykłym requestem ( bo 1000 pytań to mało efektywne) a resztę dopasowania opłaty do konkretnej ofery ( opłata związana z zamówieniem, w którym występował konkretny produkt) ( dla osoby dla której to pisze to jedno zamówienie= jeden produkt= jedna opłata związana z ofertą) Dziękuję - Temat zamknięty.
